nlp/front/storage-mongo/src/main/kotlin/UserNamespaceMongoDAO.kt

```diff
@@ -69,5 +70,5 @@ object UserNamespaceMongoDAO : UserNamespaceDAO {
         col.countDocuments(and(Login eq user, Namespace eq namespace, Owner eq true)) == 1L
 
     override fun isExistingNamespace(namespace: String): Boolean =
-        col.countDocuments(Namespace eq namespace) != 0L
+        col.countDocuments(Filters.regex(Namespace.name, "^$namespace$", "i")) != 0L
```

Comment:
I do not understand with the namespace is not case sensitive. Also this code allows injection :). 

It would be better to optionally format the namespace to lower case when the namespace is created. What is your use case ?

Reply:
yes, it's true, I can prevent this risk by using Pattern.quote : 
`val escapedNamespace = Pattern.quote(namespace)
col.countDocuments(Filters.regex(Namespace.name, "^$escapedNamespace$", "i"))`

The use case is described here : https://github.com/theopenconversationkit/tock/issues/1644

Reply:
LGTM. Yes at least quote the namespace. And add a option to disable the feature (for backward compatibility)

Reply:
If I've understood correctly, you want me to add an environment var to enable or disable this fix.
This comparison change has been made to correct an anomaly, not doing this fix means that we accept to have 'app' and 'App' as two different namespaces. So adding an option to enable or disable this comparison doesn't seem justifiable to me.

What do you think?

Reply:
Ok I understand better the use case, but :

- why changing isExistingNamespace that is not called by AdminTockUserListener ?

- you also have to change  namespaceDAO.getUsers(namespace) call (AdminTockUserListener#46) to handle case sensitive issue

Reply:
- The isExistingNamespace function is called when a namespace is created, so we check that the namespace exists before creating it.
- Yes, that's a good point. I'm working on it